File: Easy/Array/3731. Find Missing Elements/Solution.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;

        boolean[] present = new boolean[101];

        for(int num : nums){
            min = Math.min(min, num);
            max = Math.max(max, num);

            present[num] = true;
        }

        List<Integer> list = new ArrayList<Integer>();

        for(int i = min; i <= max; i++){
            if(!present[i]){
                list.add(i);
            }
        }

        return list;
    }
}

File: Easy/Array/3731. Find Missing Elements/Solution.kt
class Solution {
    fun findMissingElements(nums: IntArray): List<Int> {
        
        val min = nums.min()
        val max = nums.max()

        val set = nums.toSet()
        val list = mutableListOf<Int>()

        for(i in min..max){
            if(i !in set){
                list.add(i)
            }
        }

        return list
    }
}
